### DISCUSSION TITLE
Validating a Wireable Property

### DISCUSSION THREAD

**Discussion by stevenmond** [livewire/livewire#4032] [7. Bugs (archived)] [answered]
When I was working with validating a Livewire\Wireable object, I ran into an issue where it didn't serialize the Wireable object prior to validation resulting in it failing validation. It seemed like the function unwrapDataForValidation would basically noop it resulting in a __set_state(array()) object rather than an array to be used in validation.

Changing it from
```
protected function unwrapDataForValidation($data) {
    return collect($data)->map(function ($value) {
        if ($value instanceof Collection || $value instanceof EloquentCollection || $value instanceof Model) return $value->toArray();
        return $value;
    })->all();
}
```

to 
```
protected function unwrapDataForValidation($data)
    {
        return collect($data)->map(function ($value) {
            if ($value instanceof Collection || $value instanceof EloquentCollection || $value instanceof Model) return $value->toArray();
            else if ($value instanceof Wireable) return $value->toLivewire();
            return $value;
        })->all();
    }
```

Seemed to preliminarily fix the issue although I wasn't sure if that was correct or if I missed a serialization method that needed to be added. Thanks for your assistance.

**Comment by joshhanley** [comment, marked as answer]
@stevenmond ah yeah I don't think that was handled. Your fix seems ok to me (but I haven't looked at it in detail), would you be able to PR a fix?

Thanks!

**Reply by stevenmond** [reply]
Sure. Will submit a PR for it.
